packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js

```diff
@@ -26,36 +26,58 @@ function getPositionsOnly<T>(
   return positions.map(({ position }) => position);
 }
 
-function getAdjacentColumnHeightDeltas(heights: $ReadOnlyArray<number>): $ReadOnlyArray<number> {
-  return heights.reduce((acc, height, index) => {
+function getAdjacentColumnHeightDeltas(
+  heights: $ReadOnlyArray<number>,
+  columnSpan: number,
+): $ReadOnlyArray<number> {
+  const adjacentHeightDeltas = heights.reduce((acc: $ReadOnlyArray<number>, height, index) => {
     const adjacentColumnHeight = heights[index + 1];
     if (adjacentColumnHeight >= 0) {
       return [...acc, Math.abs(height - adjacentColumnHeight)];
     }
     return acc;
   }, []);
+
+  if (columnSpan === 2) {
+    return adjacentHeightDeltas;
+  }
+
+  return adjacentHeightDeltas.reduce((acc, _, index) => {
+    if (index + columnSpan - 2 < adjacentHeightDeltas.length) {
+      let sum = 0;
+      adjacentHeightDeltas.slice(index, index + columnSpan - 1).forEach((delta) => {
+        sum += delta;
+      });
+      return [...acc, sum / (columnSpan - 1)];
+    }
+    return acc;
+  }, []);
 }
 
-function calculateTwoColumnModuleWidth(columnWidth: number, gutter: number): number {
-  return columnWidth * 2 + gutter;
+function calculateMultiColumnModuleWidth(
```

Comment:
unrelated to this PR but also it might be good to test to see if memoizing this function might help to avoid recalculations. not urgent.

Reply:
We can verify that later